### DISCUSSION TITLE
Error when opening a model from (TMDL) folder

### DISCUSSION THREAD

**Discussion by SteiniMiracle** [TabularEditor/TabularEditor3#1136] [Other Topics]
I created a model and saved it to a folder (with serialization mode TMDL).
The  data was imported from  a SQL server with a workspace
The model worked fine and I was able to browse the data in the workspace database.

What I did when I got the error:
![image](https://github.com/TabularEditor/TabularEditor3/assets/54557808/6e66341b-813c-4c78-8dfc-53711785f766)

![image](https://github.com/TabularEditor/TabularEditor3/assets/54557808/45783fd5-b1b3-4340-9704-8d2ef209747b)
Click Yes

![image](https://github.com/TabularEditor/TabularEditor3/assets/54557808/3d759bbd-5537-47cb-a009-840c22bb45d2)
Click Ok

![image](https://github.com/TabularEditor/TabularEditor3/assets/54557808/4dcc5e3a-77e1-417a-a5cd-64fd6933178b)
Click Ok and then I get the error.

![image](https://github.com/TabularEditor/TabularEditor3/assets/54557808/8f305690-113c-46dd-8a29-09539a257585)

I'm running Windows 11 (Version 10.0.22621.1702), TE3 3.7.1 and SQL Server 2022 with latest  CU (Microsoft SQL Server 2022 (RTM-CU4) (KB5026717) - 16.0.4035.4 (X64))

I noticed that the Database-Server Name property for the model was emtpy and I was unable to enter the server name since the field is read-only.
![image](https://github.com/TabularEditor/TabularEditor3/assets/54557808/c417d40b-0f70-4aeb-9551-d9f60f2bc45e)

I hope this this is enough information to reproduce the issue (or prove me wrong).

Hilsen,
Steini

**Comment by otykier** [comment]
cc @mthierba

**Comment by otykier** [comment]
Hi @SteiniMiracle 
I believe there are still a few issues in the Preview-2 version of TMDL, which TE3 is currently using, specifically regarding certain properties on StructuredDataSource objects. This could explain the error message you're seeing, when TE3 attempts to deploy the workspace database.

Server Name will be empty by design when working offline (i.e. when loading model metadata from a file, without an attached instance of Analysis Services).

**Comment by SteiniMiracle** [comment]
Hi
Thanks for the update. I will then just wait (impatiently) for Preview-n where this will be fixed :-)

**Comment by mthierba** [comment]
@otykier Have you been able to reproduce this issue? I'd love to investigate if this is related to TMDL but I have nothing to get started with. Note that TMDL does generally not handle the Database object - that is expected to be done by the host application. TMDL's scope is from the Model down.

**Comment by SteiniMiracle** [comment]
I created a small repo with all the steps leading to the error.
Maybe it can help if you are not able to reproduce the issue.
The repo can be found at https://github.com/SteiniMiracle/TinyTest
